packages/vm/src/evm/opcodes/util.ts

```diff
@@ -261,3 +261,35 @@ export function updateSstoreGas(
     return new BN(common.param('gasPrices', 'sstoreSet'))
   }
 }
+
+export const eof1CodeAnalysis = (code: Buffer) => {
+  const secCode = 0x01
+  const secData = 0x02
+  const secTerminator = 0x00
+  let codeSize = 0
+  const sectionSizes = {
+    code: 0,
+    data: 0,
+  }
+  if (code[1] === 0x00 && code[2] === 0x01) {
+    if (code.length > 7 && code[3] === secCode && code[6] === secTerminator) {
+      codeSize = 7 + ((code[4] << 8) | code[5])
+      sectionSizes.code = (code[4] << 8) | code[5]
+    } else if (
+      code.length > 10 &&
+      code[3] === secCode &&
+      code[6] === secData &&
+      code[9] === secTerminator
+    ) {
+      codeSize = 10 + ((code[4] << 8) | code[5]) + ((code[7] << 8) | code[8])
+      sectionSizes.code = (code[4] << 8) | code[5]
+      sectionSizes.data = (code[7] << 8) | code[8]
+    }
+    if (code.length !== codeSize) {
```

Comment:
There is no check for the data size, what if this underflows or overflows?

Reply:
Ah, sorry, the naming here is imprecise. `codeSize` is the computed length of the entire bytestring based on the size of the header + the size of the code section and the data section.  I'll clean up the naming to make things a little more clear.